pkg/api/types.go

```diff
@@ -180,6 +180,150 @@ type VolumeSource struct {
 	GitRepo *GitRepoVolumeSource `json:"gitRepo"`
 	// Secret represents a secret that should populate this volume.
 	Secret *SecretVolumeSource `json:"secret"`
+	// PersistentVolumeClaimVolumeSource represents a reference to a PersistentVolumeClaim in the same namespace
+	PersistentVolumeClaimVolumeSource *PersistentVolumeClaimVolumeSource `json:"persistentVolumeClaim,omitempty"`
+}
+
+// Similar to VolumeSource but meant for the administrator who creates PVs.
+// Exactly one of its members must be set.
+type PersistentVolumeSource struct {
+	// GCEPersistentDisk represents a GCE Disk resource that is attached to a
+	// kubelet's host machine and then exposed to the pod.
+	GCEPersistentDisk *GCEPersistentDiskVolumeSource `json:"persistentDisk"`
+
+	// AWSElasticBlockStore represents an EBS volume in AWS that is attached to
+	// kubelet's host machine and then exposed to the pod.
+	AWSElasticBlockStore *AWSElasticBlockStore `json:"elasticBlockStore"`
+
+	// NFSMount represents the location of an exported volume on a fileserver
+	NFSMount *NFSMount `json:"nfsMount"`
+
+	// HostPath represents a directory on the host.
+	// This is useful for development and testing only.
+	// on-host storage is not supported in any way
+	HostPath *HostPathVolumeSource `json:"hostPath"`
+}
+
+type PersistentVolume struct {
+	TypeMeta   `json:",inline"`
+	ObjectMeta `json:"metadata,omitempty"`
+
+	//Spec defines a persistent volume owned by the cluster
+	Spec PersistentVolumeSpec `json:"spec,omitempty"`
+
+	// Status represents the current information about persistent volume.
+	Status PersistentVolumeStatus `json:"status,omitempty"`
+}
+
+type PersistentVolumeSpec struct {
+	// Resources represents the actual resources of the volume
+	Capacity ResourceList `json:"capacity`
```

Comment:
I can't leave it out here.   An admin defines a volume with a specific amount of storage (i.e, creates a GCE disk with 5gb).   PersistentVolumeSpec.Capacity["storage"] = "5GB".   This is an actual.

Compared with a PersistentVolumeClaimSpec.Resources.Request["storage"] = "3GB".   This is a request for 3gb of storage space.   PersistentVolumeClaim.Status.Capacity["storage"] = "5GB" to reflect the actual that backs their claim, because it differs from their 3gb request.